src/ui/components/LedgerAdmin.js

```diff
@@ -74,38 +101,52 @@ export const LedgerAdmin = (): ReactNode => {
     let nextLedger;
     if (ledger.account(id).active) {
       nextLedger = ledger.deactivate(id);
-      setCheckBoxSelected(false);
     } else {
       nextLedger = ledger.activate(id);
-      setCheckBoxSelected(true);
     }
     updateLedger(nextLedger);
   };
 
   const resetIdentity = () => {
     setIdentityName("");
     setSelectedId(null);
-    setCheckBoxSelected(false);
     setPromptString("Add Identity: ");
   };
 
   const setActiveIdentity = (identity: Identity) => {
     setIdentityName(identity.name);
     setSelectedId(identity.id);
-    setCheckBoxSelected(ledger.account(identity.id).active);
     setPromptString("Update Identity: ");
   };
 
+  useEffect(() => {
+    const identities = {};
+
+    ledger.accounts().map(({identity, active}) => {
+      identities[`${identity.id}`] = active;
+    });
+
+    setIsChecked(identities);
+  }, [ledger]);
```

Comment:
I recommend converting this to a `useMemo` so it's synchronous, as well as passing `ledger.accounts()` as the arg list.